serialport.c

```diff
@@ -590,6 +590,12 @@ SP_API enum sp_return sp_open(struct sp_port *port, enum sp_mode flags)
 		RETURN_CODEVAL(ret);
 	}
 
+	// When the OS gives an invalid baudrate we need to set it to a sane value
+	// See https://sigrok.org/bugzilla/show_bug.cgi?id=1632
+	if (config.baudrate == 0) {
+		config.baudrate = 9600;
```

Comment:
I think the correct thing to do here would be to put this code in `get_config()` instead, and have it set `config.baudrate` to `-1` if the reported baud rate is zero. That's the value we use to represent any unknown/invalid/unset configuration parameter - see usage notes at: https://sigrok.org/api/libserialport/unstable/a00025.html#autotoc_md13

> For each setting in a port configuration, a special value of -1 can be used, which will cause that setting to be left alone when the configuration is applied by `sp_set_config()`.
> 
> This value is also be used by `sp_get_config()` for any settings which are unconfigured at the OS level, or in a state that is not representable within the libserialport API.

Then when `set_config()` is called later in `sp_open()`, it will avoid touching the baudrate. The baud will get set when an actual requested value is provided via e.g. `sp_set_baudrate()`.

Reply:
The problem is set_config will still fail because the dcb.BaudRate is set to an invalid value of -1 (which is the same issue with value zero returned by windows itself during [GetCommState](https://github.com/sigrokproject/libserialport/blob/master/serialport.c#L1666)) which causes [SetComm](https://github.com/sigrokproject/libserialport/blob/master/serialport.c#L2023) to fail. See https://docs.microsoft.com/en-us/windows/win32/api/winbase/ns-winbase-dcb. We don't get an valid `struct sp_port *` from `sp_open` so the call to `sp_set_baudrate` is impossible.